pkg/instaclustr/client.go

```diff
@@ -718,16 +817,66 @@ func (c *Client) GetFirewallRuleStatus(
 	return firewallRuleStatus, nil
 }
 
-func (c *Client) CreateFirewallRule(
-	url string,
-	firewallRuleSpec any,
+func (c *Client) CreateClusterNetworkFirewallRule(
+	firewallRuleSpec *clusterresourcesv1beta1.ClusterNetworkFirewallRuleSpec,
+	clusterID string,
 ) (*clusterresourcesv1beta1.FirewallRuleStatus, error) {
-	jsonFirewallRule, err := json.Marshal(firewallRuleSpec)
+	payload := &struct {
+		ClusterID string `json:"clusterId"`
+		Type      string `json:"type"`
+		Network   string `json:"network"`
+	}{
+		ClusterID: clusterID,
+		Type:      firewallRuleSpec.Type,
+		Network:   firewallRuleSpec.Network,
+	}
+
+	jsonFirewallRule, err := json.Marshal(payload)
 	if err != nil {
 		return nil, err
 	}
 
-	url = c.serverHostname + url
+	url := c.serverHostname + ClusterNetworkFirewallRuleEndpoint
+	resp, err := c.DoRequest(url, http.MethodPost, jsonFirewallRule)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusAccepted {
+		return nil, fmt.Errorf("status code: %d, message: %s", resp.StatusCode, body)
+	}
+	var creationResponse *clusterresourcesv1beta1.FirewallRuleStatus
+	err = json.Unmarshal(body, &creationResponse)
+	if err != nil {
+		return nil, err
+	}
+	return creationResponse, nil
+}
+
+func (c *Client) CreateAWSSecurityGroupFirewallRule(
+	firewallRuleSpec *clusterresourcesv1beta1.AWSSecurityGroupFirewallRuleSpec,
+	clusterID string,
+) (*clusterresourcesv1beta1.FirewallRuleStatus, error) {
+	payload := &struct {
+		SecurityGroupID string `json:"securityGroupId"`
+		ClusterID       string `json:"clusterId,omitempty"`
+		Type            string `json:"type"`
+	}{
+		SecurityGroupID: firewallRuleSpec.SecurityGroupID,
+		ClusterID:       clusterID,
+		Type:            firewallRuleSpec.Type,
+	}
+
+	jsonFirewallRule, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	url := c.serverHostname + AWSSecurityGroupFirewallRuleEndpoint
```

Comment:
Why do you have 2 identical functions for firewallRules? You can use 1 for them and do not create anonymous structures that already exist in models

Reply:
we have a general fireWallRuleSpec structure but firewallRules by themselves have their specific fields in spec. I guess we already had a discussion about it

Reply:
Yea, but this case is different. Here you dont need to update entity, so just use any

Reply:
yeah, but i can't add clusterID to any